### PR TITLE
[Feature] [connector] Add Spark Ignite Source

### PR DESCRIPTION
**Pull request by zixi0825**
## Purpose of this pull request

This pull request adds spark ignite source. #1127

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by CalvinKirs — COMMENTED** [review]
You should follow the steps in the Check List to add the corresponding NOTICE and LICENSE

File: seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/batch/SparkBatchExecution.scala [hunk 1]
```diff
@@ -60,6 +60,7 @@ class SparkBatchExecution(environment: SparkEnvironment)
         SparkBatchExecution.sinkProcess(environment, sink, ds)
       })
       // }
+      environment.getSparkSession.stop()
```
Comment (SteNicholas) [inline seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/batch/SparkBatchExecution.scala:63]: Why does the session stop here?

**Comment by RickyHuo** [comment]
@CalvinKirs  if developer should update **tools/dependencies/known-dependencies.txt** or not ？

**Comment by CalvinKirs** [comment]
> @CalvinKirs if developer should update **tools/dependencies/known-dependencies.txt** or not ？

yup, checklist has corresponding docs

File: docs/en/spark/configuration/source-plugins/Ignite.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Read data from Ignite.
+
+### Options
+
+| name           | type   | required | default value |
+| -------------- | ------ | -------- | ------------- |
+| [table](#table-string)       | string | yes      | -             |
+| [config.file.path](#config.file.path-string)        | string | yes      | -         |
+
```
Comment (wuchunfu) [inline docs/en/spark/configuration/source-plugins/Ignite.md:13]: If I remember correctly `result_table_name` seems to be required, please check, thanks

File: seatunnel-connectors/seatunnel-connector-spark-ignite/pom.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-api-spark</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.ignite</groupId>
+            <artifactId>ignite-spark-2.4</artifactId>
+        </dependency>
```
Comment (wuchunfu) [inline seatunnel-connectors/seatunnel-connector-spark-ignite/pom.xml:43]: Please define this version number in parent `pom.xml` , thanks

**Comment by CalvinKirs** [comment]
I'll close it due to no feedback for a long time, if you make an update, leave a comment below and I'll reopen it

**Comment by zixi0825** [comment]
@CalvinKirs please reopen it ,thanks

**Pull request closed without merging**